### PR TITLE
Use PR source branch for build status

### PR DESCRIPTION
**Pull request by jaguilar-atl**
The new `BitbucketPullRequestSCMHead` uses `PR-{id}` as the format for the head name. This means that when the build status is posted back to bitbucket it uses that as the ref name. This PR modifies this so that it uses the PR source branch instead.

### CODE REVIEW

**Comment by jaguilar-atl** [comment]
> How does the server use the "ref" property of the build status? There is the commit ID as well, so I don't see why it needs "ref".
> 
> If there is a pull request from a fork, then the plugin posts the build status to the target repository where the pull request resides, right? In that case, is it correct to use the ref name from the source repository? The target repository might have an identically named ref with different commits, and it might not be the target branch of the pull request.

@KalleOlaviNiemitalo Thank you for your feedback.

Regarding the server's use of the "ref" property in the build status, legacy build statuses (before Bitbucket 7.14) do not require refs, as commit IDs are sufficient. However, with modern build statuses, Bitbucket utilizes the ref from the pull request's source branch (and source repo for cross-repo pull requests) as part of the criteria to find the builds to display on the pull request builds page.

As for pull requests from forks, although we are actively considering adding support for forked pull requests in future updates, we have decided not to implement it as part of the current round of changes. As mentioned, Bitbucket uses the source ref as part of the criteria to search for builds to display on the pull request page, including the source ref from the source pull request for forked repos. Therefore, when we do implement forked PR support, it is likely that the build statuses will need to be posted to the source repository.

If you have any further suggestions or questions, please feel free to share them.

**Comment by jaguilar-atl** [comment]
- [ ] Resolve conflicts with https://github.com/jenkinsci/atlassian-bitbucket-server-integration-plugin/pull/403

File: src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketPullRequestSourceBranch.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    private final MinimalPullRequest pullRequest;
+
+    public BitbucketPullRequestSourceBranch(MinimalPullRequest pullRequest) {
+        this.pullRequest = requireNonNull(pullRequest, "pullRequest");
+    }
+
+    @Override
+    public void populateEnvironmentVariables(GitSCM scm, Map<String, String> env) {
+        env.put(PULL_REQUEST_SOURCE_BRANCH, pullRequest.getFromRefDisplayId());
```
Comment (dkjellin) [inline src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketPullRequestSourceBranch.java:27]: Not opposing, just pointing out that this is now API and we cannot remove it or change format of it.

**Review by dkjellin — APPROVED** [review]

**Comment by jaguilar-atl** [comment]
> Bitbucket utilizes the ref from the pull request's source branch (and source repo for cross-repo pull requests) as part of the criteria to find the builds to display on the pull request builds page.

I'd like to clarify that this is only one of the possible criteria for displaying builds in the pull request page. Specifying the ref is recommended to allow restricting the build to a specific ref, but it's not required.

**Pull request merged**